### PR TITLE
Add support TypeScript

### PR DESCRIPTION
**Pull request by 0xlay**
Support TypeScript was added: https://github.com/notepad-plus-plus/notepad-plus-plus/issues/10353#issue-963344194

### CODE REVIEW

**Comment by donho** [comment]
the extension "ts" is recognized as JavaScript, which applies to Cpp lexer also. What makes your PR better?
A screenshot of Before and After could help.

**Comment by 0xlay** [comment]
@doho TypeScript has extensions ts, tsx. JavaScript has extensions js, jsx. These are two different languages. TypeScript has static typing, but JavaScript does not.

My PR adds TypeScript lang as a new language.

**Comment by donho** [comment]
> My PR adds TypeScript lang as a new language.

Yes, I do understand what you have added in your PR. However, besides adding  TypeScript lang, I don't see the advantage of syntax highlighting in TypeScript.

**Comment by 0xlay** [comment]
@donho I added new keywords to the "langs.model.xml" file for the TypeScript lang that are not in JavaScript. And some of the keywords I removed. I think the logical solution would be to add TypeScript as a new language, rather than mixing everything with one JavaScript. But it's decide you.

File: PowerEditor/src/Notepad_plus.cpp [hunk 1]
```diff
@@ -3448,6 +3448,8 @@ LangType Notepad_plus::menuID2LangType(int cmdID)
             return L_TXT2TAGS;
         case IDM_LANG_VISUALPROLOG:
             return L_VISUALPROLOG;
+		case IDM_LANG_TYPESCRIPT:
+			return L_TYPESCRIPT;
```
Comment (donho) [inline PowerEditor/src/Notepad_plus.cpp:3452]: Indentation should be aligned with other lines.

File: PowerEditor/visual.net/notepadPlus.vcxproj [hunk 2]
```diff
@@ -530,6 +530,11 @@ copy ..\src\contextMenu.xml ..\binarm64\contextMenu.xml
   <ItemGroup>
     <Image Include="..\src\icons\allChars_off.ico" />
     <Image Include="..\src\icons\allChars_on.ico" />
+    <Image Include="..\src\icons\asciiPanel.ico" />
+    <Image Include="..\src\icons\asciiPanel2.ico" />
+    <Image Include="..\src\icons\chameleon.ico" />
+    <Image Include="..\src\icons\clipboardPanel.ico" />
+    <Image Include="..\src\icons\clipboardPanel2.ico" />
```
Comment (donho) [inline PowerEditor/visual.net/notepadPlus.vcxproj:537]: All the diff in `notepadPlus.vcxproj` should be restored.
There's no reason to modify project file.

**Review by donho — CHANGES_REQUESTED** [review]
Please correct the 2 requested points.

**Comment by 0xlay** [comment]
@donho This project has a code style problem (mixing tabs and spaces). GitHub doesn't like the mixing of tabs and spaces.

**Review by donho — APPROVED** [review]

**Comment by donho** [comment]
@0xlay 
I just realized that in dark mode there's no syntax highlighting in TypeScript document, do you mind to add it in `DarkModeDefault.xml` please ?

**Comment by donho** [comment]
@0xlay 
Another request for auto-completion & function list of type script - but it could be done later.

**Comment by 0xlay** [comment]
@donho Ok. I will add in next PR support dark theme and auto-completion & function list of type script

**Pull request closed without merging**